railties/lib/rails/generators/app_base.rb

```diff
@@ -140,6 +141,70 @@ def build(meth, *args) # :doc:
         builder.public_send(meth, *args) if builder.respond_to?(meth)
       end
 
+      def deduce_implied_options(options, option_reasons, meta_options)
+        active = options.transform_values { |value| [] if value }.compact
+        irrevocable = (active.keys - meta_options).to_set
+
+        deduction_order = TSort.tsort(
+          ->(&block) { option_reasons.each_key(&block) },
+          ->(key, &block) { option_reasons[key]&.each(&block) }
+        )
+
+        deduction_order.each do |name|
+          reasons = option_reasons[name]&.select(&active).presence
+          active[name] ||= reasons if reasons
+          irrevocable << name if reasons&.any?(irrevocable)
+        end
+
+        revoked = options.select { |name, value| value == false }.keys.to_set - irrevocable
+        deduction_order.reverse_each do |name|
+          revoked += option_reasons[name].to_a if revoked.include?(name)
+        end
+        revoked -= meta_options
+
+        active.filter_map do |name, reasons|
+          reasons -= revoked.to_a
+          [name, reasons] unless revoked.include?(name) || reasons.empty?
+        end.to_h
+      end
+
+      OPTION_IMPLICATIONS = { # :nodoc:
+        (:skip_active_job)     => [:skip_action_mailer, :skip_active_storage],
```

Comment:
Why is this inside parentheses?

Reply:
Because I wanted to use hash rocket syntax to imitate "A ⇒ B" ("A implies B"), but RuboCop complains when the key is just a symbol.  (Without the parentheses, RuboCop wants `A: B`.)

Reply:
Not sure if using the hash rocket syntax here help. Using parentheses is just confusing. I prefer consistency over meanings that only one person have.

Reply:
I really thought those parentheses had some meaning. In the `app_generator` specially, like parentheses have meaning in routes. I think I'd go with the hash rockets, or alternatively disable the cop inline.

Reply:
I have no strong opinion about that, but I was really confused for one minute. Your call.